persistence-jdbc/core/src/main/resources/reference.conf

```diff
@@ -6,12 +6,64 @@ jdbc-defaults.slick {
   # set to one of: slick.jdbc.PostgresProfile$, slick.jdbc.MySQLProfile$, slick.jdbc.OracleProfile$ or slick.jdbc.H2Profile$
   # profile = "slick.jdbc.PostgresProfile$"
 
-  # The JNDI name
-  jndiName = "DefaultDS"
+  # The JNDI name for the Slick pre-configured DB
+  # By default, this value will be used by all akka-persistence-jdbc plugin components (journal, read-journal and snapshot).
+  # you may configure each plugin component to use different DB settings
+  jndiDbName=DefaultDB
+}
+
+
+db.default {
+
+  # The JNDI name for this DataSource
+  # Play, and therefore Lagom, will automatically register this DataSource as JNDI resource using this name.
+  # This DataSource will be used to build a pre-configured Slick DB
+  jndiName=DefaultDS
+
+  # Lagom will configure a Slick Database, using the async-executor settings below
+  # and register it as a JNDI resource using this name.
+  # By default, all akka-persistence-jdbc plugin components will use this JDNI name
+  # to lookup for this pre-configured Slick DB
+  jndiDbName=DefaultDB
+
+  async-executor {
+    # number of objects that can be queued by the async executor
+    queueSize = 10000
+
+    # 5 * number of cores
+    numThreads = 20
+
+    # number of threads
+    minConnections = 20
+
+    # 5 * number of threads
+    maxConnections = 100
+  }
+
+  # Hikari is the default connection pool and it's fine-tuned with to use the same
+  # values for minimum and maximum connections as defined for the async-executor above
+  hikaricp {
+    minimumIdle = ${db.default.async-executor.minConnections}
+    maximumPoolSize = ${db.default.async-executor.maxConnections}
+  }
+
+  # Alternatively, BoneCP can be used instead of Hikari.
+  # The settings below configured it to use the same
+  # values for minimum and maximum connections as defined for the async-executor above
+  bonecep {
+    # the pool partition count
+    partitionCount = 1
+
+    # the value below is dependent on the partitionCount
+    # it must be equal or less than async-executor.minConnections / partitionCount
+    minConnectionsPerPartition = ${db.default.async-executor.minConnections}
+
+    # the value below is dependent on the partitionCount
+    # it must be equal or less than async-executor.maxConnections / partitionCount
+    maxConnectionsPerPartition = ${db.default.async-executor.maxConnections}
+  }
```

Comment:
I'm not really convinced of this BoneCP settings. It may add more confusion to the mix. 

As it is now, it's incomplete because these settings will be ignored if the user don't swap to BoneCP. Maybe we should have documentation section about BoneCP or a recipe, but not properties that are incomplete or disabled.

Reply:
I think the comments you added are helpful. Maybe we should also link to the Play docs on configuring the connection pool implementation: https://www.playframework.com/documentation/2.6.x/ScalaDatabase#Selecting-and-configuring-the-connection-pool

I'm in favor of having the settings here because people who read those docs will expect they can just switch to BoneCP with `play.db.pool=bonecp` and all will be fine. With these default settings, it should be OK. If they want to customize things further, they can, but the comments here serve as a notice that you'll need to be thoughtful about it.

It is probably a good idea to expand our documentation on connection pool and execution context management, but for now I think this is good.